include/sparrow_v01/layout/nested_value_types.hpp

```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+#include "sparrow_v01/layout/list_layout/list_value.hpp"
+#include "sparrow_v01/layout/struct_layout/struct_value.hpp"
```

Comment:
Why do we need this header? depending on `list_value` should not require depending on `struct_value`

Reply:
tricky one! otherwise you either have to include all of these in the unit tests for lets say test_list_value (otherwise one gets an incomplete type) 
Therefore it makes sense to bundle them in one place and just include that header everywhere the variant is used

Reply:
Got it! I think it deserves a comment.